packages/ra-ui-materialui/src/AdminContext.tsx

```diff
@@ -1,16 +1,37 @@
 import * as React from 'react';
 import { CoreAdminContext, CoreAdminContextProps } from 'ra-core';
 
-import { defaultTheme } from './defaultTheme';
-import { ThemeProvider } from './layout/Theme';
+import { defaultLightTheme } from './defaultTheme';
+import { ThemeProvider, ThemesContext, RaThemeOptions } from './layout/Theme';
 
-export const AdminContext = (props: CoreAdminContextProps) => {
-    const { theme = defaultTheme, children, ...rest } = props;
+export const AdminContext = (props: AdminContextProps) => {
+    const {
+        theme,
+        lightTheme = defaultLightTheme,
+        darkTheme,
+        defaultToLightTheme,
+        children,
+        ...rest
+    } = props;
     return (
         <CoreAdminContext {...rest}>
-            <ThemeProvider theme={theme}>{children}</ThemeProvider>
+            <ThemesContext.Provider
```

Comment:
I'm not fond of this syntax (ThemesContextProvider wraps ThemeProvider), but it's a good way to make the whole change backwards compatible.

Reply:
I think the existing FIXMEs introduced in this PR will suffice